optical_rl_gym/envs/rmcsa_env.py

```diff
@@ -162,18 +174,18 @@ def reset(self, only_counters=True):
     def render(self, mode='human'):
         return
 
-    def _provision_path(self, path: Path, initial_slot, number_slots):
+    def _provision_path(self, core: int, path: Path, initial_slot, number_slots):
         # usage
-        if not self.is_path_free(path, initial_slot, number_slots):
+        if not self.is_path_free(core, path, initial_slot, number_slots):
             raise ValueError("Path {} has not enough capacity on slots {}-{}".format(path.node_list, path, initial_slot,
                                                                                      initial_slot + number_slots))
 
         self.logger.debug('{} assigning path {} on initial slot {} for {} slots'.format(self.service.service_id, path.node_list, initial_slot, number_slots))
         for i in range(len(path.node_list) - 1):
-            self.topology.graph['available_slots'][self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                                                                        initial_slot:initial_slot + number_slots] = 0
-            self.spectrum_slots_allocation[self.topology[path.node_list[i]][path.node_list[i + 1]]['index'],
-                                                    initial_slot:initial_slot + number_slots] = self.service.service_id
+            self.topology.graph['available_slots'][initial_slot:initial_slot + number_slots, core,
```

Comment:
@carlosnatalino Here's the other instance of our change.

Reply:
Note that in line 157, you define the `available_slots` as (spatial_resources, edges, spectrum_resources).

Therefore, the access to this matrix should always follow the same order. This means that the core should be indexed before the slots, i.e., `core` should come before link and slots.